src/utils/DatasetsUtils.js

```diff
@@ -9,21 +9,18 @@ import {
   CONNECTOR_NAME_ADT,
   STORAGE_ROOT_DIR_PLACEHOLDER,
 } from '../services/config/ApiConstants';
-import { SecurityUtils } from './SecurityUtils';
 import { Api } from '../services/config/Api';
 import { dispatchSetApplicationErrorMessage } from '../state/dispatchers/app/ApplicationDispatcher';
+import { ACL_ROLES } from '../services/config/accessControl';
+import { SecurityUtils } from './SecurityUtils';
 
 const patchDatasetWithCurrentUserPermissions = (dataset, userEmail, permissionsMapping) => {
-  if (dataset?.security == null || Object.keys(dataset?.security).length === 0) return;
+  let userPermissions;
+  if (dataset?.security == null)
+    userPermissions = SecurityUtils.getPermissionsFromRole(ACL_ROLES.DATASET.ADMIN, permissionsMapping);
+  else userPermissions = SecurityUtils.getUserPermissionsForResource(dataset.security, userEmail, permissionsMapping);
 
-  dataset.security = {
-    ...dataset.security,
-    currentUserPermissions: SecurityUtils.getUserPermissionsForResource(
-      dataset.security,
-      userEmail,
-      permissionsMapping
-    ),
-  };
+  dataset.security = { ...dataset.security, currentUserPermissions: userPermissions };
```

Comment:
`?` operator behind dataset ?

Reply:
Actually, if `dataset` is `null` or `undefined`, calling the function is pointless because there is no dataset to patch. I'll add a safety check at the start of the function and remove the optional chaining operators in the rest of the function code :+1:

Reply:
Fixed